### DISCUSSION TITLE
Migration from v1.6.1 or Morphia to  2.x

### DISCUSSION THREAD

**Discussion by terenceq** [MorphiaOrg/morphia#2880] [General]
### Question about remaining on v1.6.1 of Morphia

Currently I'm working on large project that leverages Morphia quite a bit. We started at several years ago with Morphia v1.3.x and then managed to migrate to v1.6.1. We are now at a place where we would like to do the following:

- Move to v4.11.x of the MongoDB Java driver
- Move from targeting Java 11 to Java 17

Are there any limitations or reasons we could not remain on v1.6.1 of Morphia given the points above?  

** Please complete the following information: **
    Server Version: v5 and v6
    Driver Version: v4.4
    Morphia Version: v1.6.1

**Comment by evanchooly** [comment]
The java version almost certainly doesn't matter.  The driver, however, has seen some evolutions that are incompatible with older code.  If I'm reading this correctly, you're moving from driver version 4.4 to 4.11.x.  That's *probably* a safe move but the Morphia 1.6 code is quite old at this point so there might be some hidden dangers in there.  I would suggest you bump both versions and run your tests and see what happens.  If it fails, you know your answer.  If the tests pass, it becomes a question of how much you trust your tests to emulate the scenarios you see in production.

**Reply by terenceq** [reply]
Thanks for the quick reply - this will help with making our decision. I am leery of the hidden dangers that you mention as well. We've already identified a couple of issues where we are broken by the move from the driver version 4.4 to 4.11 (separate from Morphia)
